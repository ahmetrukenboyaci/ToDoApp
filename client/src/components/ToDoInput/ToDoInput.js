/** Dependencies **/
import React, {useCallback} from 'react';
import PropTypes from 'prop-types';

/** Styles **/
import styles from './toDoInput.module.scss';

const ToDoInput = ({onChange, onKeyDown, newValue}) => {
	const onValueChange = useCallback((vl) => {
		onChange(vl);
	}, [onChange]);

	return (
		<div className={styles.toDoInputContainer}>
			<input
				value={newValue}
				onChange={e => onValueChange(e.target.value)}
				onKeyDown={e => (e.key === 'Enter') ? onKeyDown() : {}}
				placeholder={'Add a To Do'}
			/>
		</div>
	);
};

ToDoInput.propTypes = {
	onChange: PropTypes.func,
	onKeyDown: PropTypes.func,
	newValue: PropTypes.string
};

export default ToDoInput;