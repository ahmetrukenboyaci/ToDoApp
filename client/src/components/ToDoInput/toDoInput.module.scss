.toDoInputContainer {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 16px;
  border: 1px solid #eee;
  border-radius: 4px;
  width: 100%;

  &:focus-within {
    border: 1px solid #233184;
  }

  & input {
    border: none;
    outline: none;
    width: 100%;
  }
}