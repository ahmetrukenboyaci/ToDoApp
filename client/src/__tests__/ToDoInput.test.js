import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {configure, shallow} from 'enzyme';
import ToDoInput from '../components/ToDoInput/ToDoInput';

configure({adapter: new Adapter()});

describe('ToDoInput', () => {
	it('should render todoinput component', () => {
		// eslint-disable-next-line no-unused-vars
		const wrapper = shallow(<ToDoInput/>);
	});

	it('should render initial layout', () => {

		const component = shallow(<ToDoInput/>);

		expect(component.getElements()).toMatchSnapshot();
	});

	it('should return a value on change', () => {

		let resultValue = '';

		const mockFn = (vl) => {
			resultValue = vl;
		};

		const component = shallow(<ToDoInput onChange={mockFn}/>);
		const form = component.find('input');

		form.props().onChange({
			target: {
				value: 'value'
			}
		});

		expect(resultValue).toBe('value');
	});

	it('should call onkeydown event on enter', () => {

		let resultValue = '';

		const mockFn = () => {
			resultValue = 'entered';
		};

		const component = shallow(<ToDoInput onKeyDown={mockFn}/>);
		const form = component.find('input');

		form.props().onKeyDown({
			key: 'Enter'
		});

		expect(resultValue).toBe('entered');
	});
});