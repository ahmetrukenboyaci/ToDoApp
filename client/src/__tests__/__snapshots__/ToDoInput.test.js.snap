// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoInput should render initial layout 1`] = `
Array [
  <div
    className="toDoInputContainer"
  >
    <input
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Add a To Do"
    />
  </div>,
]
`;
